deploy/run.sh

```diff
@@ -12,22 +12,37 @@ if [ ! -x $CADVISOR_PORT ]; then
   EXTRA_ARGS="$EXTRA_ARGS --cadvisor_port=$CADVISOR_PORT"
 fi
 
+# Select what source sink to use. Options: "influxdb,gcm"
+# Default is InfluxDB.
+if [ -x $SINK ]; then
+  SINK="influxdb"
+fi
+
 HEAPSTER="/usr/bin/heapster $EXTRA_ARGS "
-# Check if InfluxDB service is running
-if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
-  # TODO(vishh): add support for passing in user name and password.    
-  INFLUXDB_ADDRESS=""
-  if [ ! -z $MONITORING_INFLUXDB_SERVICE_HOST ]; then
-    INFLUXDB_ADDRESS="${MONITORING_INFLUXDB_SERVICE_HOST}:${MONITORING_INFLUXDB_SERVICE_PORT}"
+
+if [ "$SINK" == "influxdb" ]; then
+  # Check if InfluxDB service is running
+  if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
+    # TODO(vishh): add support for passing in user name and password.    
+    INFLUXDB_ADDRESS=""
+    if [ ! -z $MONITORING_INFLUXDB_SERVICE_HOST ]; then
+      INFLUXDB_ADDRESS="${MONITORING_INFLUXDB_SERVICE_HOST}:${MONITORING_INFLUXDB_SERVICE_PORT}"
+    elif [ ! -z $INFLUXDB_HOST ]; then
+      INFLUXDB_ADDRESS=${INFLUXDB_HOST}
+    else 
+      echo "InfluxDB service address not found. Exiting."
+      exit 1
+    fi
+    $HEAPSTER --kubernetes_master "${KUBERNETES_RO_SERVICE_HOST}:${KUBERNETES_RO_SERVICE_PORT}" --sink influxdb --sink_influxdb_host $INFLUXDB_ADDRESS
   elif [ ! -z $INFLUXDB_HOST ]; then
-    INFLUXDB_ADDRESS=${INFLUXDB_HOST}
-  else 
-    echo "InfluxDB service address not found. Exiting."
-    exit 1
+    $HEAPSTER --sink influxdb --sink_influxdb_host ${INFLUXDB_HOST}
+  else
+    $HEAPSTER
+  fi
+elif [ "$SINK" == "gcm" ]; then
+  if [ ! -z $KUBERNETES_RO_SERVICE_HOST ]; then
+    $HEAPSTER --kubernetes_master "${KUBERNETES_RO_SERVICE_HOST}:${KUBERNETES_RO_SERVICE_PORT}" --sink gcm
```

Comment:
Should we do the master service discovery ahead of time and prepare heapster args before checking for storage drivers?